scanner/src/main/kotlin/storages/FileBasedStorage.kt

```diff
@@ -20,15 +20,59 @@
 
 package com.here.ort.scanner.storages
 
+import com.fasterxml.jackson.module.kotlin.readValue
+
+import com.here.ort.model.Identifier
 import com.here.ort.model.Package
+import com.here.ort.model.ScanResult
 import com.here.ort.model.ScanResultContainer
 import com.here.ort.model.ScannerDetails
+import com.here.ort.model.yamlMapper
 import com.here.ort.scanner.ScanResultsStorage
+import com.here.ort.utils.collectMessagesAsString
 import com.here.ort.utils.log
+import com.here.ort.utils.storage.FileStorage
+
+import java.io.IOException
+import java.io.PipedInputStream
+import java.io.PipedOutputStream
 
 const val SCAN_RESULTS_FILE_NAME = "scan-results.yml"
 
-abstract class FileBasedStorage : ScanResultsStorage() {
+/**
+ * A [ScanResultsStorage] using a [FileStorage] as backend. Scan results are serialized using [YAML][yamlMapper].
+ */
+class FileBasedStorage(
+    /**
+     * The [FileStorage] to use for storing scan results.
+     */
+    private val backend: FileStorage
+) : ScanResultsStorage() {
+    override fun readFromStorage(id: Identifier): ScanResultContainer {
+        val path = storagePath(id)
+
+        @Suppress("TooGenericExceptionCaught")
+        return try {
+            backend.read(path).use { input ->
+                yamlMapper.readValue(input)
+            }
+        } catch (e: Exception) {
+            when (e) {
+                is java.lang.IllegalArgumentException, is IOException -> {
```

Comment:
Maybe using FileNotFoundException could be an option?

Reply:
`FileNotFoundException` is a subclass of `IOException`, so it is already handled.

Reply:
I meant changing the called `fun` to throw only `FileNotFoundE...` but not `IllegalArgument...`
anyhow, I'm ok with it as is as well.